test/TEST-21-DFUZZER/test.sh

```diff
@@ -12,6 +11,13 @@ TEST_FORCE_NEWIMAGE=1
 # shellcheck source=test/test-functions
 . "${TEST_BASE_DIR:?}/test-functions"
 
+# Run the test either only under nspawn (if $TEST_PREFER_NSPAWN is set to true)
+# or only uder qemu otherwise, to avoid running the test twice on machines where
+# we can do both.
+if ! get_bool "${TEST_PREFER_NSPAWN:=}"; then
+    TEST_NO_NSPAWN=1
+fi
```

Comment:
As far as I understand `TEST_NO_QEMU` should be set to `1` if `TEST_PREFER_NSPAWN` is `0`

Reply:
Looks like `do_test` takes care of that

Reply:
@mrc0mmand as far as I can tell if `TEST_NO_QEMU=yes` is set explicitly the test is skipped completely without `TEST_PREFER_NSPAWN` and that's weird in the sense that I'm not sure why I should pass
```
TEST_PREFER_NSPAWN=yes
```
if I already pass `TEST_NO_QEMU=yes`. I think I'll go ahead and merge the PR once the CI is green but all those variables like `TEST_PREFER_NSPAWN` that haven't been integrated into the testsuite properly should probably be revisited.